c++/src/kj/memory.h

```diff
@@ -313,6 +313,14 @@ class Own<T, decltype(nullptr)> {
   inline operator T*() { return ptr; }
   inline operator const T*() const { return ptr; }
 
+  // Surrenders ownership of the underlying object to the caller. The caller must pass in the 
+  // correct disposer to prove that they know how the object is meant to be disposed of. 
+  inline T* disown(const Disposer* d) {
+    KJ_REQUIRE(d == disposer, "When disowning an object, disposer must be equal to Own's disposer");
```

Comment:
KJ_REQUIRE isn't defined here, and you probably can't #include it without creating a cyclic dependency. You will need to do something like:

`if (d != disposer) throwWrongDisposerError();`

and then implement `throwWrongDisposerError()` out-of-line (in `memory.c++`).